crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/CalculateContributionService.java

```diff
@@ -31,93 +30,215 @@ public class CalculateContributionService {
+        if (compareContributionService.compareContribution(calculateContributionDTO) < 2) {
+            return maatCourtDataService.createContribution(createContributionRequest, laaTransactionId);
+        } else return null;
     }
 
-    public CalculateContributionResponse calcContribs(ContributionDTO contributionDTO, ContributionResponseDTO contributionResponseDTO, String laaTransactionId) {
+    public CalculateContributionResponse calcContribs(final CalculateContributionDTO calculateContributionDTO,
+                                                      final ContributionResponseDTO contributionResponseDTO,
+                                                      final String laaTransactionId) {
         CalculateContributionResponse response = new CalculateContributionResponse();
-        LocalDate assEffectiveDate = getEffectiveDate(contributionDTO);
+        LocalDate assEffectiveDate = getEffectiveDate(calculateContributionDTO);
         ContributionCalcParametersDTO contributionCalcParametersDTO = maatCourtDataService.getContributionCalcParameters(assEffectiveDate.toString(), laaTransactionId);
-        CrownCourtOutcome crownCourtOutcome = contributionRulesService.getActiveCCOutcome(contributionDTO.getCrownCourtSummary());
-        boolean isContributionRuleApplicable = contributionRulesService.isContributionRuleApplicable(contributionDTO.getCaseType(),
-                contributionDTO.getMagCourtOutcome(), crownCourtOutcome);
+        CrownCourtOutcome crownCourtOutcome = contributionRulesService.getActiveCCOutcome(calculateContributionDTO.getCrownCourtSummary());
+        boolean isContributionRuleApplicable = contributionRulesService.isContributionRuleApplicable(calculateContributionDTO.getCaseType(),
+                calculateContributionDTO.getMagCourtOutcome(), crownCourtOutcome);
 
-        BigDecimal annualDisposableIncome = calculateAnnualDisposableIncome(contributionDTO, laaTransactionId, crownCourtOutcome, isContributionRuleApplicable);
+        BigDecimal annualDisposableIncome = calculateAnnualDisposableIncome(calculateContributionDTO, laaTransactionId, crownCourtOutcome, isContributionRuleApplicable);
 
         if (contributionResponseDTO.getUpliftCote() != null &&
-                contributionDTO.getDateUpliftApplied() != null &&
-                contributionDTO.getDateUpliftRemoved() == null) {
+                calculateContributionDTO.getDateUpliftApplied() != null &&
+                calculateContributionDTO.getDateUpliftRemoved() == null) {
             BigDecimal monthlyContributions = calculateUpliftedMonthlyAmount(annualDisposableIncome, contributionCalcParametersDTO);
             response.setMonthlyContributions(monthlyContributions);
             response.setUpliftApplied(Constants.Y);
-        } else if (contributionResponseDTO.getCalcContribs().equals(Constants.N)) {
+        } else if (Constants.N.equals(contributionResponseDTO.getCalcContribs())) {
```

Comment:
calContribs() is called when contributionResponseDTO.getCalcContribs() is equal to 'Y'. Line 191 will never be executed. Procedure also has the same logic but I feet it is a redundant logic?